packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1760,28 +1760,22 @@ mixin WidgetInspectorService {
     return _safeJsonEncode(_getChildrenSummaryTree(diagnosticsNodeId, groupName));
   }
 
-  DiagnosticsNode? _idToDiagnosticsNode(String? diagnosticsOrDiagnosticableId) {
-    // TODO(polina-c): start always assuming Diagnosticable, when DevTools stops sending DiagnosticsNode to
-    // APIs that invoke this method.
-    // https://github.com/flutter/devtools/issues/3951
-    final Object? object = toObject(diagnosticsOrDiagnosticableId);
+  DiagnosticsNode? _diagnosticableIdToDiagnosticsNode(String? diagnosticableId) {
```

Comment:
nit: consider leaving this method name alone. The new one is quite long. You could change the parameter to a required named parameter and that would clarify that the input must be a diagnosticableId.